File: Scripts/utils/config_utils.lua
local configUtils = { }
local language_file

function configUtils.GetFeatureSetting(section, setting)
    local featuresSettings = JsonRead("Settings\\feature_settings.json")
    
    return featuresSettings[section][setting]
end

function configUtils.SetFeatureSetting(section, setting, value)
    local featuresSettings = JsonRead("Settings\\feature_settings.json")
    
    featuresSettings[section][setting] = value

    JsonSave("feature_settings.json", featuresSettings)
end

function configUtils.GetCurrentLanguage()
    local config = JsonRead("Settings\\config.json")
        
    if config["language"] then
        return config["language"]
    else
        return ""
    end
end
            
function configUtils.GetString(key)
    if language_file == nil then
        language_file = JsonRead("Language\\" .. configUtils.GetCurrentLanguage() .. ".json")
    end

    if language_file[key] then
        return language_file[key]
    else
        return ""
    end
end

return configUtils

File: Scripts/utils/config_utils.py
def get_feature_settings():
    import GCT
    import json
    
    with open(GCT.GetGCTFolder() + "\\Settings\\feature_settings.json", "r", encoding="utf-8") as f:
        settings = json.load(f)
    
    return settings
    
def save_feature_settings(settings):
    import GCT
    import json
    
    with open(GCT.GetGCTFolder() + "\\Settings\\feature_settings.json", "w", encoding="utf-8") as f:
        settings = json.dump(settings, f, indent=4)
        
def GetCurrentLanguage():
    import GCT
    import json
    
    with open(GCT.GetGCTFolder() + "\\Settings\\config.json", "r", encoding="utf-8") as f:
        config = json.load(f)
        
    if "language" in config:
        return config["language"]
    else:
        return ""
            
def GetString(key : str):
    import GCT
    import json
    
    global language_file
    
    try:
        if key in language_file:
            return language_file[key]
        else:
            return ""
    except NameError:
        with open(GCT.GetGCTFolder() + f"\\Language\\{GetCurrentLanguage()}.json", "r", encoding="utf-8") as f:
            language_file = json.load(f)
        return GetString(key)
        
        